File: Vlad-and-an-Odd-Ordering.py
# Author : ysh
# 2024/02/22 Thu 17:51:46
# https://codeforces.com/contest/1926/problem/E
import sys
input = sys.stdin.readline
print = sys.stdout.write

def check(n,x):
    pre = (n + 1) >> 1
    if x - pre <= 0:
        return (x << 1) - 1
    return (check(n >> 1,x - pre) << 1)

for i in range(int(input())):
    a,b = [int(i) for i in input().split()]
    print(f'{check(a,b)}\n')

File: Vlad-and-an-Odd-Ordering.kt
// Author : ysh
// 2024/02/22 Thu 18:01:57
// https://codeforces.com/contest/1926/problem/E
fun check(n: Int,x: Int): Int {
    var pre = (n + 1) / 2
    if(x - pre <= 0) return (x * 2) - 1
    return (check(x / 2,x - pre) * 2)
}
fun main() = repeat(readLine()!!.toInt()) {
    var (a,b) = readLine()!!.split(' ').map {x -> x.toInt()}
    println("${check(a,b)}")
}